lib/phoenix/live_dashboard/helpers/supervision_tree_helpers/reingold_tifford_algorithm.ex

```diff
@@ -0,0 +1,150 @@
+defmodule Phoenix.LiveDashboard.ReingoldTilford do
+  # Reingold-Tilford algorithm for drawing trees
+  @moduledoc false
+
+  @height 20
+  @node_y_separation 10
+  @total_y_distance @height + @node_y_separation
+
+  def set_layout_settings(tree) do
+    tree
+    |> calculate_initial_y(0, [])
+    |> ensure_children_inside_screen()
+    |> put_final_y_values(0)
+  end
+
+  defp calculate_initial_y(%{children: children} = node, previous_sibling, top_siblings) do
+    {_, children} =
+      children
+      |> Enum.reduce({0, []}, fn n, {prev_sibling, nodes} ->
+        new_node = calculate_initial_y(n, prev_sibling, nodes)
+        {new_node.y, nodes ++ [new_node]}
+      end)
```

Comment:
Instead of appending to the list, which copies the whole list on every operation, we generally prefer to prepend:

```suggestion
    {_, reversed_children} =
      Enum.reduce(children, {0, []}, fn n, {prev_sibling, nodes} ->
        new_node = calculate_initial_y(n, prev_sibling, nodes)
        {new_node.y, [new_node | nodes]}
      end)

    [last_child | _] = reversed_children
    [first_child | _] = children = Enum.reverse(reversed_children)
```

The benefit of this approach is that we get the first and last children for free, so you no longer need to do `List.last` later on.